Clean SWIG bindings + add doc

cchampet opened the pull request.

Commit:
Item: ignore methods which return boost objects in bindings

Commit:
Sequence: ignore methods which return boost object in bindings

Commit:
ItemStat: ignore constructor with a boost object as parameter in bindings

Commit:
Sequence: getFiles method returns a list of string instead of bfs::path

* Can be used in bindings.
* The methods is not called in the library.

Commit:
Item: added doc to getTypeFromPath method

Commit:
Sequence: added doc to public methods

Commit:
Sequence: added getLastFilename method

To keep symmetry with getFirstFilename method.

Commit:
USAGE.md: added more examples to manipulate sequences

Commit:
FrameRange: removed SWIG warnings

Renamed operators in bindings.

Commit:
Item: removed SWIG warnings

Commit:
Sequence: removed SWIG warnings

Commit:
common: removed SWIG warnings

Commit:
CMake: updated SWIG flags to handle warning as error

The compilation will failed if SWIG has warnings when generate bindings.

Commit:
Item: added doc to the class

Commit:
USAGE: added a note to explain the difference between Item and Sequence

fabiencastan commented on src/sequenceParser/Sequence.hpp:
```diff
@@ -127,6 +129,7 @@ class Sequence
 
 	inline std::string getSuffix() const;
 
+#ifndef SWIG
```
Why do you want to remove this function from the binding?

cchampet replied:
Found a way to integrate this method to the SWIG interfaces: 66e7002

fabiencastan commented on src/sequenceParser/Sequence.hpp:
```diff
@@ -206,28 +239,12 @@ class Sequence
-	 */
+	std::string _prefix;
+	std::string _suffix;
+
```
why do you remove the maxPadding doc?

cchampet replied:
I moved it to the getMaxPadding method. See commit c1560e42004d8fb46685239ff4676d6614b5881b

fabiencastan commented on USAGE.md:
```diff
@@ -1,13 +1,9 @@
-Just folders.
-
+* Sequence: many files with a common pattern of numbers.
+* File: all file without numbers in the filename and sequences with only one file.
```
The description was already bad, but maybe we can clean it a bit.
We should not use "sequence" to define a "file".
"sequences with only one file" is a really bad wording because by default in sequenceparser a sequence contains at least 2 files.

Proposition:

```
* Sequence: a collection of files with a common prefix and suffix around a varying number.
* File: a file alone which is not included in a file sequence
* Folder: All folders. Folders are never considered as a sequence.
```

cchampet replied:
Agreed: the definition are not good. See commit d0cd3b9

fabiencastan commented on src/sequenceParser/FrameRange.i:
```diff
@@ -14,6 +14,8 @@
         def __str__(self):
             return self.string()
+        def __eq__(self, other):
+            return self.first == other.first and self.last == other.last and self.step == other.step
```
You can call the c++ code ouside of the pythoncode:

```
  bool __eq__(const FrameRange& o) const {
    return *$self == o;
  }
```

cchampet replied:
Even better here eb7cfaa :)

fabiencastan commented on src/sequenceParser/Sequence.i:
```diff
@@ -20,11 +20,15 @@ namespace std {
+                    return self.string()
+                def __lt__(self, other):
+                    return self.getFilenameWithStandardPattern() < other.getFilenameWithStandardPattern()
+                def __eq__(self, other):
```
it's a bit dangerous to have multiple comparison code (one  in c++ and one in python).

cchampet replied:
A better solution here cb6ba7e

Commit:
USAGE: updated definitions section

Commit:
FrameRange: clean FrameRange class

Separate methods from attributes, even if both are public.

Commit:
Item: renamed operators in bindings

Instead of redefine them in Python, and ignore them in Java.

Commit:
Sequence: renamed operators in bindings

Instead of redefine them in Python, and ignore them in Java.

Commit:
USAGE: renamed local variables when get general info of a sequence

To understand that Time = Frame.

cchampet commented:
@fabiencastan Ready to be merged ;)

Commit:
Sequence: fixed return value of getGlobalRange method in SWIG interfaces

* Instanciate tempate type "TimePair".
* Updated SWIG type "Time" to match C++ type (fixed the following SWIG
error: "cannot convert ‘std::pair<int, int>*’ to ‘std::pair<long int,
long int>*’ in assignment").

Commit:
Sequence: added isIn method to the SWIG interfaces

* Need to appy typemaps to handle output parameters.
* Need to define a custom typemap for std::string.

Commit:
Sequence: updated isIn method

* Initialize the output arguments.
* Needed when calling this method from a SWIG interface.

Commit:
Sequence: added check of the time of the given filename in isIn method

Commit:
pyTest: added a test to check Sequence::isIn method
